src/components/MoveModal.vue

```diff
@@ -0,0 +1,269 @@
+					<a @click.prevent="onClickCrumb(index)">
+						{{ getMailboxTitle(box) }}
+					</a>
+				</div>
+			</span>
+			<div class="mailbox-list">
+				<ul v-if="mailboxes.length > 0">
+					<li
+						v-for="box in mailboxes"
+						:key="box.databaseId"
+						@click.prevent="onClickMailbox(box)">
+						<div :class="['mailbox-icon', getMailboxIcon(box)]" />
+						<div class="mailbox-title">
+							{{ getMailboxTitle(box) }}
+						</div>
+					</li>
+				</ul>
+				<div v-else class="empty">
+					<div class="empty-icon icon-folder" />
+					<h2>{{ t('mail', 'No more subfolders in here') }}</h2>
+				</div>
+			</div>
+			<div class="buttons">
+				<button class="primary" :disabled="!isMoveable" @click="onMove">
+					<span v-if="moving" class="icon-loading-small spinner" />
+					{{ moving ? t('mail', 'Moving') : t('mail', 'Move') }}
+				</button>
+			</div>
+		</div>
+	</Modal>
+</template>
+
+<script>
+import Modal from '@nextcloud/vue/dist/Components/Modal'
+
+import { translate as translateMailboxName } from '../i18n/MailboxTranslator'
+
+export default {
+	name: 'MoveModal',
+	components: {
+		Modal,
+	},
+	props: {
+		envelopes: {
+			type: Array,
+			required: true,
+		},
+	},
+	data() {
+		return {
+			moving: false,
+			hasMovedEnvelopes: false,
+			destMailboxId: undefined,
+			mailboxCrumbs: [],
+		}
+	},
+	computed: {
+		account() {
+			return this.$store.getters.getAccount(this.envelopes[0].accountId)
+		},
+		isMoveable() {
+			return !this.moving && this.destMailboxId
+		},
+		mailboxes() {
+			let mailboxes = []
+			if (!this.destMailboxId) {
+				mailboxes = this.$store.getters.getMailboxes(this.account.accountId)
+			} else {
+				mailboxes = this.$store.getters.getSubMailboxes(this.destMailboxId)
+			}
+			return mailboxes
+		},
+	},
+	methods: {
+		onClose() {
+			this.$emit('close')
+			if (this.hasMovedEnvelopes) {
+				this.$store.dispatch('syncEnvelopes', { mailboxId: this.destMailboxId })
+			}
+		},
+		onClickHome() {
+			this.destMailboxId = undefined
+			this.mailboxCrumbs = []
+		},
+		onClickCrumb(index) {
+			this.destMailboxId = this.mailboxCrumbs[index].databaseId
+			this.mailboxCrumbs = this.mailboxCrumbs.slice(0, index + 1)
+		},
+		onClickMailbox(mailbox) {
+			this.destMailboxId = mailbox.databaseId
+			this.mailboxCrumbs.push(mailbox)
+		},
+		onMove() {
+			if (!this.isMoveable) {
+				return
+			}
+
+			this.moving = true
+
+			const envelopeIds = this.envelopes
+				.filter((envelope) => envelope.mailboxId !== this.destMailboxId)
+				.map((envelope) => envelope.databaseId)
+
+			if (envelopeIds.length === 0) {
+				this.$emit('close')
+				return
+			}
+
+			const promises = envelopeIds.map(async(id) => {
+				await this.$store.dispatch('moveMessage', {
+					id,
+					destMailboxId: this.destMailboxId,
+				})
+				this.hasMovedEnvelopes = true
+			})
+
+			Promise.all(promises).then(() => {
+				this.$store.dispatch('syncEnvelopes', { mailboxId: this.destMailboxId })
```

Comment:
@st3iny : 

error "Could not sync envelopes: Request failed with status code 428 Error: Request failed with status code 428" most likely occurs here

Reply:
'don't have the error anymore....